src/backend/renderer/gles/mod.rs

```diff
@@ -3054,8 +3097,13 @@ impl<'frame> GlesFrame<'frame> {
 
 impl<'frame> Drop for GlesFrame<'frame> {
     fn drop(&mut self) {
-        if let Err(err) = self.finish_internal() {
-            warn!("Ignored error finishing GlesFrame on drop: {}", err);
+        match self.finish_internal() {
+            Ok(sync) => {
+                sync.wait();
```

Comment:
Is this is going to still cause GlesFrame to block even the SyncPoint is used in KMS?